service-front/module/Application/src/Helpers/DependencyCheck.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Application\Helpers;
+
+use Aws\Ssm\SsmClient;
+use Application\Enums\IdMethod;
+
+class DependencyCheck
+{
+    public function __construct(private readonly array $statusData, private array $processedStatus = [])
```

Comment:
Similarly, some more specific type hints would help any future refactoring:
```suggestion
    /**
     * @param array<string, bool>
     * @return array<string, bool>
     */
    public function __construct(private readonly array $statusData, private array $processedStatus = [])
```